design/src/rfcs/rfc0017_logging_sensitive.md

```diff
@@ -0,0 +1,376 @@
+# RFC: Logging in the Presence of Sensitive Data
+
+> Status: RFC
+
+Smithy provides a [sensitive trait](https://awslabs.github.io/smithy/1.0/spec/core/documentation-traits.html#sensitive-trait) which exists as a `@sensitive` field annotation syntactically and has the following semantics:
+
+> Sensitive data MUST NOT be exposed in things like exception messages or log output. Application of this trait SHOULD NOT affect wire logging (i.e., logging of all data transmitted to and from servers or clients).
+
+This RFC is concerned with solving the problem of honouring this specification.
+
+Progress has been made towards this goal in the form of the [Sensitive Trait PR](https://github.com/awslabs/smithy-rs/pull/229), which uses code generation to remove sensitive fields from `Debug` implementations.
+
+The problem remains open due to the existence of HTTP binding traits and a lack of clearly defined user guidelines which customers may follow to honour the specification.
+
+This RFC proposes a new logging `Layer` to be generated and applied to each `OperationHandler` `Service`, and internal and external developer guidelines on how to avoid violating the specification.
```

Comment:
I would give some details about the fact that our users also have no understanding of the routing decisions before they happen and this can be very confusing and hard to debug.

Remember this applies only to `rest*` protocols. AwsJson1.0 and 1.1 can be already enhanced with logging information about the URI (always /), the method (always POST) and the operation (x-amzn-target header) without caring about the sensitive trait since all the fields are part of the payload, so routing info are safe.